src/deploy/functions/release/index.ts

```diff
@@ -144,3 +141,119 @@ export function printTriggerUrls(results: backend.Backend): void {
     logger.info(clc.bold("Function URL"), `(${getFunctionLabel(httpsFunc)}):`, httpsFunc.uri);
   }
 }
+
+/**
+ * Checks if artifact cleanup policies are set for the regions where functions are deployed
+ * and automatically sets up policies where needed.
+ *
+ * The policy is only set up when:
+ * 1. No cleanup policy exists yet
+ * 2. No other cleanup policies exist (beyond our own if we previously set one)
+ * 3. User has not explicitly opted out
+ */
+async function checkArtifactCleanupPolicies(
+  projectId: string,
+  endpoints: backend.Endpoint[],
+): Promise<void> {
+  if (endpoints.length === 0) {
+    return;
+  }
+
+  const uniqueRegions = new Set<string>();
+  for (const endpoint of endpoints) {
+    uniqueRegions.add(endpoint.region);
+  }
+
+  const regionResults = await Promise.all(
+    Array.from(uniqueRegions).map(async (region) => {
+      try {
+        const repoPath = artifacts.makeRepoPath(projectId, region);
+        const repository = await artifactregistry.getRepository(repoPath);
+        const existingPolicy = artifacts.findExistingPolicy(repository);
+        const hasPolicy = !!existingPolicy;
+        const hasOptOut = artifacts.hasCleanupOptOut(repository);
+
+        // Check if there are any other cleanup policies beyond our own
+        const hasOtherPolicies =
+          repository.cleanupPolicies &&
+          Object.keys(repository.cleanupPolicies).some(
+            (key) => key !== artifacts.CLEANUP_POLICY_ID,
+          );
+
+        return {
+          region,
+          repository,
+          hasPolicy,
+          hasOptOut,
+          hasOtherPolicies,
+        };
+      } catch (err) {
+        logger.debug(`Failed to check artifact cleanup policy for region ${region}:`, err);
+        return {
+          region,
+          hasPolicy: false,
+          hasOptOut: false,
+          hasOtherPolicies: false,
+          error: err,
+        };
+      }
+    }),
+  );
+
+  const regionsWithErrors = regionResults
+    .filter((result) => result.error)
+    .map((result) => result.region);
+
+  const regionsToSetup = regionResults.filter(
+    (result) => !result.hasPolicy && !result.hasOptOut && !result.hasOtherPolicies && !result.error,
+  );
+
+  const regionsNeedingWarning: string[] = [];
+
+  if (regionsToSetup.length > 0) {
+    utils.logLabeledSuccess(
+      "functions",
+      `Configuring a cleanup policy for repositories in ${regionsToSetup.join(", ")}. ` +
+        `Images older than ${artifacts.DEFAULT_CLEANUP_DAYS} days will be automatically deleted.`,
```

Comment:
It gives me the heebies to depend on a const like this inside the function. Can we at least make it a parameter and default it to the const?

Also, why use the default instead of prompting with this as the default value of the prompt?

Reply:
Interesting. I've never even considered introducing a prompt at the end of the deployment, thinking that it's an abrupt change in behavior of how functions deployment has worked for a long time.

Happy to make it a prompt and refactor things a bit to share more code between this and the command if you like the idea of prompting at the end.